spark/src/main/java/org/apache/iceberg/actions/Actions.java

```diff
@@ -77,6 +82,120 @@ public ExpireSnapshotsAction expireSnapshots() {
     return new ExpireSnapshotsAction(spark, table);
   }
 
+  /**
+   * Converts the provided table into an Iceberg table in place. The table will no longer be represented by it's
+   * previous provider in the session catalog and a new metadata directory will be created at the table's location.
+   *
+   * @param tableName Table to be converted
+   * @return {@link CreateAction} to perform migration
+   */
+  public static CreateAction migrate(String tableName) {
```

Comment:
Well, it seems we rely too much on reflection here. When we decided to use reflection to split actions, we did not have these static methods and it was reasonable since the scope of the change was smaller compared to introducing `BaseActions`. I am no longer sure reflection is a good idea here as making these methods work with reflection is more complicated than having `BaseActions`. On top, we don't have compile time checks. Users will call these methods in Spark 2 and will get runtime exceptions. I'd prefer to not expose those methods at all.

Reply:
What are your thoughts on this, @RussellSpitzer @rdblue?

Reply:
I think not implementing these in Spark 2 is definitely a reason to reconsider this.

What about somewhere in the middle? We could introduce an API that isn't static, then call its methods from the static ones here. Then we just need an implementation class, which we could load dynamically.

Reply:
I still prefer having BaseActions with an Actions per module. I think having two classes each which determine their implementation at runtime is not much better than all the static method reflection currently in the PR. The user experience would be worse, since there is a weird middle method to call, and we still would have runtime exceptions for bad method calls.

Maybe I don't understand the API that isn't static plan.

Reply:
I think the last time we chatted having `SparkActions` or move current actions to `org.apache.iceberg.spark` was one of the most promising ideas due to its simplicity. 

I agree it is not something we should address in this PR but I'd try to solve it before 0.11.